tests/base/product_type_04.cc

```diff
@@ -0,0 +1,67 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2015 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+// test that the product between a tensor and an integer behaves just
+// like that of the tensor and the integer-converted-to-double
+
+#include "../tests.h"
+#include <iomanip>
+#include <iomanip>
+#include <fstream>
+#include <cmath>
+#include <typeinfo>
+#include <complex>
+
+#include <deal.II/base/template_constraints.h>
+#include <deal.II/base/tensor.h>
+
+
+
+int main()
+{
+  std::ofstream logfile("output");
+  deallog.attach(logfile);
+  deallog.depth_console(0);
+  deallog.threshold_double(1.e-10);
+
+  // try it for a rank-1 tensor
+  {
+    Tensor<1,3> t;
+    t[0] = 1;
+    t[1] = 2;
+    t[2] = 3;
+    Assert (7*t == 7.0 * t, ExcInternalError());
+    Assert (t*7 == t * 7.0, ExcInternalError());
+    Assert (t*7 == 7*t    , ExcInternalError());
+    Assert (t*7 == t+t+t+t+t+t+t, ExcInternalError());
+  }
+
```

Comment:
Don't you want to use AssertThrow ? This test does nothing in release mode, doesn't it ?